packages/shared/src/services/coingecko/CoingeckoClient.ts

```diff
@@ -49,6 +50,20 @@ export class CoingeckoClient {
     }
   }
 
+  async getImageUrl(id: CoingeckoId): Promise<string> {
+    const data = await this.query(`/coins/${id.toString()}`, {
+      localization: 'false',
+      tickers: 'false',
+      market_data: 'false',
+      community_data: 'false',
+      developer_data: 'false',
+      sparkline: 'false',
+    })
+
+    const parsed = CoinMetadata.parse(data)
+    return parsed.image.large
```

Comment:
I remember we had a conversation that we want "all the icon sizes", what has changed since then?

Reply:
It's trivial to go from large -> small/thumb with a simple `.replace()`.